src/main/clojure/clara/rules/engine.cljc

```diff
@@ -29,12 +29,12 @@
 ;; Token with no bindings, used as the root of beta nodes.
 (def empty-token (->Token [] {}))
 
-;; Record indicating the negation existing in the working memory.
-(defrecord NegationResult [gen-rule-name])
+;; A marker interface to identify internal facts.
+(definterface ISystemFact)
```

Comment:
True, but realistically I don't expect that to change and tests would fail immediately if it did when we failed to match on NegationResult facts in extracted rules.  To be honest I see that as one of those areas of Clojure where it isn't an "official" API but in practice it is; maybe there are contrary conversations on that that I haven't seen though.  Reader conditionals would work though albeit a bit awkwardly; I'm fine with that approach.